synapse/handlers/admin.py

```diff
@@ -29,10 +32,106 @@
 class AdminHandler(BaseHandler):
     def __init__(self, hs):
         super(AdminHandler, self).__init__(hs)
-
         self.storage = hs.get_storage()
         self.state_store = self.storage.state
 
+    async def validate_admin_token(
```

Comment:
this should go in `synapse.api.auth` alongside the existing functions that handle auth tokens.